include/swift/AST/DiagnosticsParse.def

```diff
@@ -129,12 +129,19 @@ ERROR(lex_invalid_unicode_scalar,none,
 ERROR(lex_unicode_escape_braces,none,
       "expected hexadecimal code in braces after unicode escape", ())
 ERROR(lex_illegal_multiline_string_start,none,
-      "invalid start of multi-line string literal", ())
+      "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "invalid end of multi-line string literal", ())
-ERROR(lex_ambiguous_string_indent,none,
-      "invalid mix of multi-line string literal indentation", ())
-
+      "multi-line string literal content must not be on the delimiter's line", ())
+ERROR(lex_insufficient_string_indent,none,
+      "string literal content is insufficiently indented", ())
+ERROR(lex_unexpected_tab_in_string_indent,none,
+      "string literal content is indented with a tab where a space is expected", ())
+ERROR(lex_unexpected_space_in_string_indent,none,
+      "string literal content is indented with a space where a tab is expected", ())
+NOTE(note_change_current_line_indentation,none,
+      "change indentation to match last line", ())
```

Comment:
How would you feel about: "change _this line's_ indentation to match last line"--not just for symmetry, but so that the message doesn't only mention the last line when the key thing is that we're offering to change the current line?